src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -0,0 +1,63 @@
+import {
+	CourseInfoContainer,
+	GoBackLink,
+	InfoWrapper,
+	Description,
+	InfoField,
+	Key,
+} from './CourseInfo.styles';
+import { CustomTitle } from '../../common/CustomTitle/CustomTitle';
+
+import { useContext } from 'react';
+import { CoursesContext } from '../../helpers/context/courses.context';
+import { AuthorsContext } from '../../helpers/context/authors.context';
+
+import { toHoursAndMinutes } from '../../helpers/timeConvert';
+
+import { useParams } from 'react-router-dom';
+
+const CourseInfo = () => {
+	const { coursesList } = useContext(CoursesContext);
+	const { authorsList } = useContext(AuthorsContext);
+
+	const { courseId } = useParams(); //or useLocation?
+
+	const { id, creationDate, description, duration, title, authors } =
+		coursesList.find((course) => course.id === courseId);
```

Comment:
This code causes bugs. If you go to your app and write `/courses/283468264` all app is crashing 
![image](https://user-images.githubusercontent.com/44047007/227162820-7fbefd0a-5543-4efc-bfc1-c715bf90eb22.png)

Reply:
No problem, I added a button. I also tied clear form fields to this button, if not needed, I'll remove it.

As for the error, it's because I don't have a 404 page.
And I don't know how we will solve this further in the tasks.
If I now solve this by redirecting to the courses page?
Or create some kind of PageNotFound.
then the solution is this
<Route path="*" element={<PageNotFound />} />

Or should I check if the courseInfo page can be rendered before attempting to render it?

Reply:
Basically, you just need to create a fallback to check if you have a course id or not. in the end, you can either write a 404 page or just redirect to '/courses' (the second one is the easiest, so let's go with it)